scheduler/replica_scheduler.go

```diff
@@ -97,3 +100,28 @@ func (rcs *ReplicaScheduler) getNodeInfo() (map[string]*longhorn.Node, error) {
 	}
 	return scheduledNode, nil
 }
+
+func (rcs *ReplicaScheduler) generateReplica(replica *longhorn.Replica, node *longhorn.Node) error {
+	labels := replica.GetLabels()
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	if labels[types.LonghornNodeKey] == "" {
```

Comment:
You can add the label in the datastore since it's always the same as the node id. And it's datastore responsible for find the replicas on the certain node. Try to keep logic self-contained.